### ISSUE TITLE
[quotes] Insert unexpected quotes when using this plugin and format code by prettier.

### ISSUE THREAD

**Issue by zhengxiaoyao0716** [typescript-eslint/typescript-eslint#546]
**Repro**
```JSON
{
  "extends": [
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier"
  ],
  "rules": {
    "quotes": ["error", "single"]
  }
}
```

```TS
import React from 'react';

const test1 = (
  <>
    <div />
  </>
);

const test2 = (
  <div>
    <div />
  </div>
);

```

**Expected Result**
All works well.

**Actual Result**
Each time format the code, then it would insert unexpected quotes between `<><div/></>`, looks like this:
![image](https://user-images.githubusercontent.com/12682063/58157810-b89c1780-7cab-11e9-91c8-11ab8f00dc49.png)

**Additional Info**
To reproduce it, maybe we should notice that:
1. we must both using `plugin:react/recommended` `@typescript-eslint/eslint-plugin` and `prettier`.
2. `quotes` rule must be set, the inserted quotes would be `'` if `quotes` rule set to `single`, and be `"` if set to `double`.
3. it would only insert unexpected quotes  between `<></>` (short syntax of `fragments`), as the above code `test1` shows.
4. it would only happend in `.tsx` files, I tried `jsx` but everything works well inconceivable.

**Versions**

| package                            | version   |
| ---------------------------------- | --------- |
| `@typescript-eslint/eslint-plugin` | `1.9.0`   |
| `@typescript-eslint/parser`        | `1.9.0`   |
| `TypeScript`                       | `3.4.5`   |
| `ESLint`                           | `5.16.0`  |
| `node`                             | `v8.11.3` |
| `npm`                              | `6.8.0`   |

**Comment by bradzacher** [comment]
what command are you running to cause this to happen?
`eslint --fix`? `prettier --write`?

**Comment by zhengxiaoyao0716** [comment]
I'm using `VSCode` and format codes by [`Prettier` plugin](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode).

**Comment by bradzacher** [comment]
Are you using the eslint integration option within the plugin? if so then yes, that is known to be broken.

**Comment by zhengxiaoyao0716** [comment]
Yes, I set the `"prettier.eslintIntegration"` option of the plugin setting to `true`. then what should I do now?

**Comment by bradzacher** [comment]
You should not use that option.
It's broken because the plugin uses ESLint v4 (and is hard locked to v4). This plugin *requires* v5.

Subscribe to https://github.com/prettier/prettier-vscode/issues/763 and wait for a fix.

Until then you can work around this by installing the eslint extension and turning on the eslint extension's autofix on save setting.
